lib/src/async/future_stream.dart

```diff
@@ -56,31 +54,31 @@ class FutureStream<T> extends Stream<T> {
 
   static T _identity<T>(T t) => t;
 
-  Future<Stream<T>> _future;
-  StreamController<T> _controller;
-  StreamSubscription<T> _subscription;
+  late final Future<Stream<T>> _future;
+  StreamController<T>? _controller;
+  StreamSubscription<T>? _subscription;
 
   void _onListen() {
     _future.then((stream) {
       if (_controller == null) return;
-      _subscription = stream.listen(_controller.add,
-          onError: _controller.addError, onDone: _controller.close);
+      _subscription = stream.listen(_controller!.add,
+          onError: _controller!.addError, onDone: _controller!.close);
     });
   }
 
   void _onCancel() {
-    if (_subscription != null) _subscription.cancel();
+    if (_subscription != null) _subscription?.cancel();
```

Comment:
Why even do the null test when you can just make this:

`_subscription?.cancel();`

Reply:
I'll go hang my head in shame now. I bet I did that in stream_router too.

Reply:
Done.

Reply:
<3